File: TransCoder_tokenized_test_set_functions/cpp/MAXIMUM_SUM_PATH_MATRIX_TOP_BOTTOM.cpp
int maxSum ( int mat [ SIZE ] [ SIZE ], int n ) {
  if ( n == 1 ) return mat [ 0 ] [ 0 ];
  int dp [ n ] [ n ];
  int maxSum = INT_MIN, max;
  for ( int j = 0;
  j < n;
  j ++ ) dp [ n - 1 ] [ j ] = mat [ n - 1 ] [ j ];
  for ( int i = n - 2;
  i >= 0;
  i -- ) {
    for ( int j = 0;
    j < n;
    j ++ ) {
      max = INT_MIN;
      if ( ( ( j - 1 ) >= 0 ) && ( max < dp [ i + 1 ] [ j - 1 ] ) ) max = dp [ i + 1 ] [ j - 1 ];
      if ( ( ( j + 1 ) < n ) && ( max < dp [ i + 1 ] [ j + 1 ] ) ) max = dp [ i + 1 ] [ j + 1 ];
      dp [ i ] [ j ] = mat [ i ] [ j ] + max;
    }
  }
  for ( int j = 0;
  j < n;
  j ++ ) if ( maxSum < dp [ 0 ] [ j ] ) maxSum = dp [ 0 ] [ j ];
  return maxSum;
}

File: TransCoder_tokenized_test_set_functions/python/MAXIMUM_SUM_PATH_MATRIX_TOP_BOTTOM.py
def maxSum ( mat, n ) :
    if n == 1 :
        return mat [ 0 ] [ 0 ]
    dp = [ [ 0 for i in range ( n ) ] for i in range ( n ) ]
    maxSum = INT_MIN
    for j in range ( n ) :
        dp [ n - 1 ] [ j ] = mat [ n - 1 ] [ j ]
    for i in range ( n - 2, - 1, - 1 ) :
        for j in range ( n ) :
            maxi = INT_MIN
            if ( ( ( ( j - 1 ) >= 0 ) and ( maxi < dp [ i + 1 ] [ j - 1 ] ) ) ) :
                maxi = dp [ i + 1 ] [ j - 1 ]
            if ( ( ( ( j + 1 ) < n ) and ( maxi < dp [ i + 1 ] [ j + 1 ] ) ) ) :
                maxi = dp [ i + 1 ] [ j + 1 ]
            dp [ i ] [ j ] = mat [ i ] [ j ] + maxi
    for j in range ( n ) :
        if ( maxSum < dp [ 0 ] [ j ] ) :
            maxSum = dp [ 0 ] [ j ]
    return maxSum
